File: packages/server/app/models/points_delta.py
from typing import Literal

from pydantic import BaseModel

PointsDeltaReason = Literal["purchase", "return", "promo", "adjustment"]
PointsDeltaStatus = Literal["pending", "applied"]


class PointsDelta(BaseModel):
    type: Literal["points_delta"] = "points_delta"
    delta_id: str
    user_id: str
    txn_id: str
    delta: int
    reason: PointsDeltaReason
    store_id: str
    status: PointsDeltaStatus = "pending"
    created_at: str

File: packages/shared/src/models/points-delta.ts
export type PointsDeltaReason = "purchase" | "return" | "promo" | "adjustment";
export type PointsDeltaStatus = "pending" | "applied";

export interface PointsDelta {
  type: "points_delta";
  delta_id: string;
  user_id: string;
  txn_id: string;
  delta: number;
  reason: PointsDeltaReason;
  store_id: string;
  status: PointsDeltaStatus;
  created_at: string;
}
